skimage/feature/peak.py

```diff
@@ -92,6 +95,19 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     array([[10, 10, 10]])
 
     """
+    def get_high_intensity_peaks(image, mask, num_peaks):
+        """
+        Return the highest intensity peak coordinates.
+        """
+        # get coordinates of peaks
+        coord = np.transpose(mask.nonzero())
+        # select num_peaks peaks
+        if coord.shape[0] > num_peaks:
+            intensities = image.flat[np.ravel_multi_index(coord.transpose(),
+                                                          image.shape)]
+            idx_maxsort = np.argsort(intensities)[::-1]
+            coord = coord[idx_maxsort][:num_peaks]
+        return coord
```

Comment:
You shouldn't need the `row_stack` call?

btw I was completely unaware of `row_stack` and `column_stack`! So handy! I can't tell you how many times I've used `newaxis` with `hstack`, which is super-annoying!

At any rate, here's some nice middle ground, seeing the issue of having to stack one way or another:

```
coord = np.transpose(np.nonzero(mask))
if coord.shape[0] > num_peaks:
    intensities = image[tuple(coord.T)]
    top_idxs = np.argsort(intensities)[-num_peaks:]
    coord = coord[top_idxs]
```